cli/util/git.go

```diff
@@ -28,23 +29,43 @@ func GetRepoName(repoUrl string) (string, error) {
 	return repoName, nil
 }
 
-// getRepoName finds upstream repo name from a given repo directory
-func GetRepoUrl(dir string) (string, error) {
+// GetRepoUrlAndRootPath finds upstream repo URL and the root local path
+func GetRepoUrlAndRootPath(dir string) (string, string, error) {
 	opt := &git.PlainOpenOptions{DetectDotGit: true}
 	r, err := git.PlainOpenWithOptions(dir, opt)
 	if err != nil {
-		return "", fmt.Errorf("error opening git dir %s: %w", dir, err)
+		return "", "", fmt.Errorf("error opening git dir %s: %w", dir, err)
 	}
+
+	repoRootPath := ""
+	repoURL := ""
 	rm, err := r.Remote(defaultRemote)
 	if err != nil {
-		return "", fmt.Errorf("error finding remote %s in git dir %s: %w", defaultRemote, dir, err)
+		return repoURL, repoRootPath, fmt.Errorf("error finding remote %s in git dir %s: %w", defaultRemote, dir, err)
 	}
 
+	if len(rm.Config().URLs) > 0 {
+		repoURL = resolveRemoteSSHURLToHTTPS(rm.Config().URLs[0])
+	}
+
+	if repoURL == "" {
+		return repoURL, repoRootPath, fmt.Errorf("empty URL")
+	}
+
```

Comment:
nit: could this be simplified to
```suggestion
	if len(rm.Config().URLs) < 1 {
		return repoURL, repoRootPath, fmt.Errorf("no remote urls")
	}
        repoURL = resolveRemoteSSHURLToHTTPS(rm.Config().URLs[0])

```

Reply:
ack

Reply:
Actually, we can't because it still returns an empty URL if there isn't a remote. The only way to check is to assert an empty string.